python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +221,90 @@ def execute(self, settings: CLISettings) -> None:
+        Command line arguments, SignalType, hashes, and ambiguous SignalTypes are validated.
+        Returns a dict of validated SignalType, and validated hashes of each SignalType.
+        """
+
+        # There is a signal without a prefix and it's ambiguous (fix: -S)
+        if None in hashes_grouped_by_prefix:
+            if len(hashes_grouped_by_prefix) > 2 or len(signal_types) > 1:
+                # Ambiguous -- throw
+                raise CommandError(
+                    f"Error: The SignalType is ambiguous for '{self.content_type.get_name()}'"
+                    "For '--hashes' please use '-S' or '--only-signal' to specify one of"
+                    f"{[s.get_name() for s in signal_types]}",
+                    2,
+                )
+            # unambiguous -- set
+            hashes_grouped_by_prefix.setdefault(
+                signal_types[0].get_name(), set()
+            ).update(hashes_grouped_by_prefix.pop(None))
+
+        # Validate that the SignalTypes and hashes are valid (fix: remove or update hash)
+        hashes_grouped_by_signal = self.validate_signal_type_and_hash(
+            settings, hashes_grouped_by_prefix
+        )
+
+        # There is a signal not valid for your arguments (fix: remove or change args)
+        if self.only_signal and (
+            self.only_signal not in hashes_grouped_by_signal
+            or len(hashes_grouped_by_signal) > 1
+        ):
+            raise CommandError(
+                f"Error: SignalType '{self.only_signal} was specified, but attempting to query with additional/different SignalTypes"
+                f"Please remove or change your arguments, or remove the additional SignalTypes"
+            )
+
+        return hashes_grouped_by_signal
+
+    def validate_signal_type_and_hash(
+        self,
+        settings: CLISettings,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]],
+    ) -> t.Dict[t.Type[SignalType], t.Set[str]]:
+        hashes_grouped_by_signal: t.Dict[t.Type[SignalType], t.Set[str]] = {}
+        for possible_signal, hashes in hashes_grouped_by_prefix.items():
+            if not possible_signal:
+                # This shouldn't be hit as we filter out the None key before
+                continue
+            # Validate the signal
+            try:
+                signal_type = settings.get_signal_type(possible_signal)
+            except:
+                logging.exception("Signal type %s is invalid.", possible_signal)
+                raise CommandError(
+                    f"Error: '{possible_signal}' is not a valid Signal Type."
+                    "Please remove or update this hash from your query",
+                    2,
+                )
+            # Validate the hashes
+            for hash in hashes:
+                try:
+                    hash = signal_type.validate_signal_str(hash.strip())
+                except:
```

Comment:
blocking: This will catch CTRL+C interrupt and out of memory, which I don't think you intended. use `except Exception` to narrow down to the usual set.